fairseq/modules/gumbel_vector_quantizer.py

```diff
@@ -160,6 +160,7 @@ def forward(self, x, produce_targets=False):
         avg_probs = torch.softmax(
             x.view(bsz * tsz, self.groups, -1).float(), dim=-1
         ).mean(dim=0)
+        avg_probs = avg_probs.detach()
```

Comment:
i think this might break the diversity penalty (it is applied on prob perplexity below that uses avg_probs)

Reply:
ah.. I'll remove.